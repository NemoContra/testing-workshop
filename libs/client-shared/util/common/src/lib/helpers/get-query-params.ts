import { inject } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { map, Observable } from 'rxjs';

export const getQueryParam = (key: string): Observable<string | null> =>
  inject(ActivatedRoute).queryParamMap.pipe(
    map((paramMap) => paramMap.get(key))
  );
