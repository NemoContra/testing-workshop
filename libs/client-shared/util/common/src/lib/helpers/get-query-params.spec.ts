import { Component } from '@angular/core';
import { getQueryParam } from './get-query-params';
import { createComponentFactory } from '@ngneat/spectator/jest';
import { ActivatedRoute } from '@angular/router';
import { of } from 'rxjs';

@Component({
  template: `<span>{{ test$ | async }}</span>`,
})
class TestComponent {
  test$ = getQueryParam('test');
}

describe('GetQueryParam', () => {
  const createComponent = createComponentFactory({
    component: TestComponent,
    providers: [
      {
        provide: ActivatedRoute,
        useValue: {
          queryParamMap: of(new Map([['test', 'eintest']])),
        },
      },
    ],
  });

  it('should resolve the queryParam', async () => {
    const spectator = createComponent();
    expect(spectator.fixture).toMatchSnapshot();
  });
});
