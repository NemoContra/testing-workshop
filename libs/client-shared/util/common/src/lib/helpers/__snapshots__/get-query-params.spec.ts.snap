// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetQueryParam should resolve the queryParam 1`] = `
<ng-component
  __ngContext__="0"
  test$={[Function Observable]}
>
  <span>
    eintest
  </span>
</ng-component>
`;
